CJM-67029: Adding schema and mixin for Engageable Profiles

medhakumari opened the pull request:
Please link to the issue #…

Commit:
CJM-67029: Adding schema and mixin for Engageable Profiles

CJM-67029: Adding schema and mixin for Engageable Profiles

Commit:
CJM-67029: Add status hideen

ankita1289 commented on extensions/adobe/experience/engageable-profiles.example1.json:
```diff
@@ -0,0 +1 @@
+{}
```
can we add a valid example to explain this?

medhakumari replied:
Added

Commit:
CJM-67029-added-example

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+  "description": "A Record based Engageable class for creating Engageable Profiles schemas",
+  "definitions": {
+    "engageableProfiles": {
+      "properties": {}
```
what's the purpose for keeping an empty properties list?

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+      "$ref": "#/definitions/engageableProfiles"
+    }
+  ],
+  "examples": [{}]
```
Can we fill in this example as well?

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+  ],
+  "$id": "https://ns.adobe.com/experience/customerJourneyManagement/engageableProfiles",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Engageable Profiles Fields",
```
Can we think of a more descriptive title.

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/experience/customerJourneyManagement/EngageableProfiles",
```
need to change to https://ns.adobe.com/experience/

ankita1289 replied:
Rename to EngageableProfile instead of EngageableProfiles

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Engageable Profiles Class",
+  "type": "object",
+  "meta:extensible": false,
```
TBD

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+  },
+  "required": [
+    "xdm:identityMap",
+    "https://ns.adobe.com/experience/engageableProfiles/engagement/latestTimestamp"
```
as discussed, engagement is to group the fields together, for easy selection of all fields.

ankita1289 commented on extensions/adobe/experience/engageable-profiles-class.schema.json:
```diff
@@ -0,0 +1,38 @@
+  "meta:extends": ["https://ns.adobe.com/xdm/data/record"],
+  "description": "A Record based Engageable class for creating Engageable Profiles schemas",
+  "definitions": {
+    "engageableProfiles": {
```
rrename to engageableProfile

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/experience/customerJourneyManagement/engageableProfiles",
```
remove customerJourneyManagement from path

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+  "title": "Engageable Profiles Fields",
+  "description": "A set of meta-data fields related to engageable profiles.",
+  "type": "object",
+  "meta:extensible": true,
```
TBD why extensible is true and do we need a status hidden for mixin as well? or status should be stable.

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": [
+    "https://ns.adobe.com/experience/customerJourneyManagement/engageableProfiles"
```
remove customerJourneyManagement from path

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+  "definitions": {
+    "entities": {
+      "properties": {
+        "https://ns.adobe.com/experience/customerJourneyManagement/entities/engagement": {
```
remove customerJourneyManagement from path

ankita1289 commented on extensions/adobe/experience/engageable-profiles-mixins.schema.json:
```diff
@@ -0,0 +1,50 @@
+          "description": "Fields defined for engagement",
+          "properties": {
+            "https://ns.adobe.com/experience/entities/solution": {
+              "title": "Solution/Product",
```
avoid using / because its a title. ex: "Product code" or "Solution name"

ankita1289 commented on extensions/adobe/experience/engageable-profiles.example1.json:
```diff
@@ -0,0 +1,20 @@
+      }
+    ]
+  },
+  "https://ns.adobe.com/experience/engageableProfiles": {
```
rename path

ankita1289 commented on extensions/adobe/experience/engageable-profiles.example1.json:
```diff
@@ -0,0 +1,20 @@
+  },
+  "https://ns.adobe.com/experience/engageableProfiles": {
+    "xdm:engagement": {
+      "xdm:solution": "ajo",
```
Shouldn't 'solution' be a list?? as a profile can engage with multiple solutions at a time.

ankita1289 reviewed (changes requested):
Change path and other minor comments

Commit:
CJM-67029: Adding review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

Commit:
CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

CJM-67029: Added review changes

Commit:
CJM-67029: Removed unnecessary nesting

medhakumari committed:
Merge branch 'master' into CJM-67029-engageable-profiles-schema
